### ISSUE TITLE
Generator interfaces for other methods

### ISSUE THREAD

**Issue by tejasmanohar** [jepcastelein/marketo-rest-python#6]
I think ideally all the methods should have generator interfaces. Maybe this is achievable through standardizing how they paginate Marketo API (at least those with paging tokens) and make a wrapper that implements each method with `yield`?

**Comment by jepcastelein** [comment]
agree. I only recently added it for "Get Multiple Leads by List Id", but all 'GET' requests which can return 300+ records should ideally have a Generator option. I'm open to any thoughts on how it can best be implemented and whether the average Python developer is familiar enough with Generators (I certainly hope so, just not sure).

**Comment by tejasmanohar** [comment]
not concerned with familiarity. usage is simple.

**Comment by jepcastelein** [comment]
added generators for "Get Lead Activities" and "Get Multiple Leads by Program Id". When I have time I'll add more or feel free to do a Pull Request
